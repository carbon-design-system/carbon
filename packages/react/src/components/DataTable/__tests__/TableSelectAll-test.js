/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import { Table, TableHead, TableRow, TableSelectAll } from '../';

describe('DataTable.TableSelectAll', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      id: 'id',
      name: 'select-all',
      checked: false,
      onSelect: jest.fn(),
      className: 'custom-class-name',
    };
  });

  it('should render', () => {
    const wrapper = mount(
      <Table>
        <TableHead>
          <TableRow>
            <TableSelectAll {...mockProps} />
          </TableRow>
        </TableHead>
      </Table>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with the provided class name', () => {
    const customClassName = 'custom-table-select-all-classname';
    const wrapper = mount(
      <Table>
        <TableHead>
          <TableRow>
            <TableSelectAll {...mockProps} className={customClassName} />
          </TableRow>
        </TableHead>
      </Table>
    );
    const elements = wrapper.find(`th.${customClassName}`);
    expect(elements.length).toBe(1);
  });

  it('should invoke `onSelect` when clicked', () => {
    const wrapper = mount(
      <Table>
        <TableHead>
          <TableRow>
            <TableSelectAll {...mockProps} />
          </TableRow>
        </TableHead>
      </Table>
    );
    wrapper.find('input').simulate('click');
    expect(mockProps.onSelect).toHaveBeenCalledTimes(1);
  });
});
