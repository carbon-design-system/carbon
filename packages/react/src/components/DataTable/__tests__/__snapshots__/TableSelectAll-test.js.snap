// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableSelectAll should render 1`] = `
<Table
  isSortable={false}
  overflowMenuOnHover={true}
>
  <div
    className="bx--data-table-content"
  >
    <table
      className="bx--data-table bx--data-table--no-border"
    >
      <TableHead>
        <thead>
          <TableRow>
            <tr>
              <TableSelectAll
                ariaLabel="Select all rows in the table"
                checked={false}
                className="custom-class-name"
                id="id"
                name="select-all"
                onSelect={[MockFunction]}
              >
                <th
                  className="bx--table-column-checkbox custom-class-name"
                  scope="col"
                >
                  <ForwardRef(InlineCheckbox)
                    ariaLabel="Select all rows in the table"
                    checked={false}
                    id="id"
                    name="select-all"
                    onClick={[MockFunction]}
                  >
                    <InlineCheckbox
                      ariaLabel="Select all rows in the table"
                      checked={false}
                      id="id"
                      innerRef={null}
                      name="select-all"
                      onChange={[Function]}
                      onClick={[MockFunction]}
                    >
                      <input
                        checked={false}
                        className="bx--checkbox"
                        id="id"
                        name="select-all"
                        onChange={[Function]}
                        onClick={[MockFunction]}
                        type="checkbox"
                      />
                      <label
                        aria-label="Select all rows in the table"
                        className="bx--checkbox-label"
                        htmlFor="id"
                      />
                    </InlineCheckbox>
                  </ForwardRef(InlineCheckbox)>
                </th>
              </TableSelectAll>
            </tr>
          </TableRow>
        </thead>
      </TableHead>
    </table>
  </div>
</Table>
`;
